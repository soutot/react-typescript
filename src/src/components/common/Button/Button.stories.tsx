import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import Button from './Button'
import Template from '../../Template/Template'
import i18n from '../../../i18n/i18n'

storiesOf('Button', module).add('default', () => (
  <Template>
    <Button text={i18n.t('continue')} onClick={action('Clicked')} />
  </Template>
))
