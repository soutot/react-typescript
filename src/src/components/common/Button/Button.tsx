import * as React from 'react'
import styled from 'styled-components'
import { Button } from '@material-ui/core'

import Text from '../Text/Text'
import Icon from '../Icon/Icon'
import theme from '../../../theme/theme'

// **************************************************
// *****  Styled Components
// **************************************************
const ButtonStyled = styled(Button)`
  && {
    text-align: center;
    border: 1px solid ${theme.colors.primary};
    border-radius: 20px;
    height: 40px;
    min-width: 260px;
    background: ${theme.colors.none};
    outline: none;

    &:hover {
      background: ${theme.colors.primary};
    }
  }
`

// **************************************************
// ***** Interfaces
// **************************************************
interface Props {
  text: string
  onClick?: () => void
  icon?: string
}

// **************************************************
// ***** Component
// **************************************************
const ButtonComponent = ({ text, onClick, icon, ...props }: Props) => (
  <ButtonStyled onClick={onClick} {...props}>
    <Text>{text}</Text>
    {icon && <Icon name={icon} />}
  </ButtonStyled>
)

export default ButtonComponent
