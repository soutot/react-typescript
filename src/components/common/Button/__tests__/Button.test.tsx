import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Button from '../Button'

describe('Button (Snapshot)', () => {
  it('should render Button component', () => {
    const component = renderer.create(<Button text={''} />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})

describe('Button Click', () => {
  const text = 'Button'
  const onClick = jest.fn()
  const button = mount(<Button text={text} onClick={onClick} />)

  it('Click requires onClick prop', () => {
    expect(button.props().onClick).toBeDefined()
  })

  it('should renders button', () => {
    const buttonElement = button.find('button')
    expect(buttonElement).toBeDefined()
  })

  it('Button click calls onClick', () => {
    const buttonElement = button.find('button')
    buttonElement.simulate('click')
    // expect(onClick).toHaveBeenCalled()
    expect(onClick.mock.calls.length).toEqual(1)
  })
})
