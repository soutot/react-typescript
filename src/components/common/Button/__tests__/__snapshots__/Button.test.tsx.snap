// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button (Snapshot) Button renders hello world 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat jepUei"
  disabled={false}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    <span
      className="sc-bdVaJa fdVuct"
      color={undefined}
      fontSize={undefined}
    >
      ButtonText
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;

exports[`Button (Snapshot) should render Button component 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat jepUei"
  disabled={false}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    <span
      className="sc-bdVaJa fdVuct"
      color={undefined}
      fontSize={undefined}
    >
      
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;
